### ISSUE TITLE
Simple Conditional INN

### ISSUE THREAD

**Issue by alokwarey** [VLL-HD/FrEIA#98]
I am trying to run the basic example as a Conditional INN by using the labels in the moons dataset as the condition. How do I specify the cond index and cond_shape? 

Keep getting errors: either zero-dimensional tensor (at position 1) cannot be concatenated or Tensors must have same number of dimensions: got 2 and 1

I tried several permutations, example passing the condition y as a list of tensors etc.  but none of them work.

BATCHSIZE = 100
N_DIM = 2

def subnet_fc(dims_in, dims_out):
    return nn.Sequential(nn.Linear(dims_in, 512), nn.ReLU(),
                         nn.Linear(512,  dims_out))

inn = Ff.SequenceINN(N_DIM)
for k in range(8):
    inn.append(Fm.AllIInOneBlock, subnet_constructor=subnet_fc,permute_soft=True, **cond = 0, cond_shape = (1,)**)

optimizer = torch.optim.Adam(inn.parameters(), lr=0.001)

for i in range(1000):
    optimizer.zero_grad()
    # sample data from the moons distribution
    data, label = make_moons(n_samples=BATCHSIZE, noise=0.05)
    x = torch.Tensor(data)
    **y = torch.Tensor(label.reshape(-1,1))** **(is this correct?)**
    **z, log_jac_det = inn(x, c=y)**
    loss = 0.5*torch.sum(z**2, 1) - log_jac_det
    loss = loss.mean() / N_DIM
    loss.backward()
    optimizer.step()

z = torch.randn(BATCHSIZE, N_DIM)
**samples, _ = inn(z, c=y, rev=True)**

**Comment by fdraxler** [comment]
I think you have to pass `c=[y]` as `c: Iterable[Tensor]` assumes *a list* of all conditions. I agree that this is not obvious and in many cases, there is only one condition. I hope this fixes your issue. If not, feel free to re-open it!

**Comment by alokwarey** [comment]
It worked! Thanks so much!
